internal/cmd/options.go

```diff
@@ -0,0 +1,271 @@
+/*
+Copyright 2022 The OpenVEX Authors
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/openvex/go-vex/pkg/vex"
+	"github.com/spf13/cobra"
+)
+
+type vexDocOptions struct {
+	DocumentID string
+	Author     string
+	AuthorRole string
+}
+
+const (
+	productLongFlag = "product"
+	vulnLongFlag    = "vuln"
+)
+
+// Validate checks that the document options are valid
+func (do *vexDocOptions) Validate() error {
+	if do.Author == "" {
+		return fmt.Errorf("document author cannot be blank")
+	}
+	return nil
+}
+
+func (do *vexDocOptions) AddFlags(cmd *cobra.Command) {
+	cmd.PersistentFlags().StringVar(
+		&do.DocumentID,
+		"id",
+		"",
+		"ID string for the new VEX document (autogenerated by default)",
+	)
+
+	cmd.PersistentFlags().StringVar(
+		&do.Author,
+		"author",
+		vex.DefaultAuthor,
+		"author to record in the new document",
+	)
+
+	cmd.PersistentFlags().StringVar(
+		&do.AuthorRole,
+		"author-role",
+		vex.DefaultRole,
+		"optional author role to record in the new document",
+	)
+}
+
+type vexStatementOptions struct {
+	Status          string
+	StatusNotes     string
+	Justification   string
+	ImpactStatement string
+	Vulnerability   string
+	ActionStatement string
+	Product         string
+	Subcomponents   []string
+}
+
+// Validate checks that the statement options are coherent
+func (so *vexStatementOptions) Validate() error {
+	if so.Status == string(vex.StatusAffected) &&
+		so.ActionStatement != vex.NoActionStatementMsg &&
+		so.ActionStatement != "" {
+		return errors.New("action statement can only be set when stats = \"affected\" ")
+	}
```

Comment:
Isn't this condition the expected use case for the affected status? Status is "affected", and a non-empty action statement is used?

Reply:
also nit: s/stats/status/

Reply:
Ah thanks for catching the flipped logic, fixed 👍